sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -126,12 +127,14 @@ object ResolveDefaultColumns {
    *
    * @param field         represents the DEFAULT column value whose "default" metadata to parse
    *                      and analyze.
+   * @param metadataKey   which key to look up from the column metadata; generally either
+   *                      CURRENT_DEFAULT_COLUMN_METADATA_KEY or EXISTS_DEFAULT_COLUMN_METADATA_KEY.
    * @param statementType which type of statement we are running, such as INSERT; useful for errors.
    * @return Result of the analysis and constant-folding operation.
    */
-  def analyze(field: StructField, statementType: String): Expression = {
+  def analyze(field: StructField, metadataKey: String, statementType: String): Expression = {
```

Comment:
```suggestion
  def analyze(field: StructField, statementType: String, metadataKey: String = CURRENT_DEFAULT_COLUMN_METADATA_KEY): Expression = {
```

Reply:
After this change, we can avoid update all the places that use `CURRENT_DEFAULT_COLUMN_METADATA_KEY` in default.

Reply:
Sounds good, done.